src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5815,6 +5815,20 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
                 HashSet<DiagnosticInfo> useSiteDiagnostics = null;
                 OverloadResolution.MethodInvocationOverloadResolution(methodGroup.Methods, methodGroup.TypeArguments, actualArguments, overloadResolutionResult, ref useSiteDiagnostics, isMethodGroupConversion, allowRefOmittedArguments);
                 diagnostics.Add(expression, useSiteDiagnostics);
+
+                if (overloadResolutionResult.Succeeded)
+                {
+                    var receiverParameter = overloadResolutionResult.BestResult.Member.Parameters.FirstOrDefault();
+                    Debug.Assert(receiverParameter != null);
+
+                    if (receiverParameter.RefKind == RefKind.Ref)
+                    {
+                        // If this was a ref extension method, the receiver argument must be checked for L-value constraints.
+                        // This helper method will also replace the symbol with a BadExpressionSymbol if it was invalid.
```

Comment:
Will correct the comment to be more accurate. Its correct name is `BoundBadExpression`.

---
In reply to: [120717002](https://github.com/dotnet/roslyn/pull/19703#discussion_r120717002) [](ancestors = 120717002)